### PR TITLE
hostmetricsreceiver doc updates and logging

### PR DESCRIPTION
**Pull request by jrcamp**
CPU and disk metrics are not available in gopsutils on Mac when cgo is disabled
(https://github.com/shirou/gopsutil/blob/master/cpu/cpu_darwin_nocgo.go)

Errors were not logged which lead to confusion about why things weren't
working. @james-bebbington was this intentional or unintentional? It does
spam the logs a bit (every collection interval) but I don't think we can fully
rely on users checking traces for errors.

### CODE REVIEW

**Comment by james-bebbington** [comment]
Yea I was being very conservative re logging. This probably makes sense, given that typical scrape intervals are once every 10s-5m.

If we do this, should we add the logs to the [scraper controller](https://github.com/open-telemetry/opentelemetry-collector/blob/master/receiver/receiverhelper/scrapercontroller.go#L197) instead so that errors are logged by default for all scrapers?

---

_Also note to maintainers: please merge https://github.com/open-telemetry/opentelemetry-collector/pull/1949 before this PR 😄_

**Comment by james-bebbington** [comment]
Also, should we add `// +build !darwin` to `cpu_scraper_test.go` then, and maybe disable the scraper (i.e. report an error if you try to use it) on Mac as well?

File: receiver/hostmetricsreceiver/hostmetrics_receiver.go [hunk 1]
```diff
@@ -172,7 +174,9 @@ func (hmr *receiver) scrapeMetrics(ctx context.Context) {
 	}
 
 	if len(errors) > 0 {
-		span.SetStatus(trace.Status{Code: trace.StatusCodeDataLoss, Message: fmt.Sprintf("Error(s) when scraping metrics: %v", componenterror.CombineErrors(errors))})
+		combinedErrs := componenterror.CombineErrors(errors)
+		hmr.logger.Error("Error(s) when scraping metrics", zap.Error(combinedErrs))
```
Comment (bogdandrutu) [inline receiver/hostmetricsreceiver/hostmetrics_receiver.go:178]: Why this change?

**Review by bogdandrutu — APPROVED** [review]

File: receiver/hostmetricsreceiver/hostmetrics_receiver.go (hunk 1 above)
Comment (james-bebbington) [reply receiver/hostmetricsreceiver/hostmetrics_receiver.go:178]: This branch is out of date. This file no longer exists and the trace reporting was moved to `obsreport_scraper.go`

**Comment by tigrannajaryan** [comment]
@jrcamp can you rebase?

**Review by james-bebbington — APPROVED** [review]
LGTM

Two thoughts that don't need to be addressed in this PR:

- It might be useful to make this logging optional?
- Can you create an issue for disabling the cpu scraper on darwin

**Comment by jrcamp** [comment]
> LGTM
> 
> Two thoughts that don't need to be addressed in this PR:
> 
> * It might be useful to make this logging optional?
> * Can you create an issue for disabling the cpu scraper on darwin

Ah yeah. Created #2143. Looks like disk also has the same problem (updated readme).

**Comment by james-bebbington** [comment]
> Would this be an option in ScraperControllerSettings?

I was thinking it's something developers would want to configure per scraper type rather than users configuring this directly, but maybe.
